tests/filtering/test_filter_against_latest_blocks.py

```diff
@@ -16,9 +16,8 @@ def test_filter_against_latest_blocks(web3_empty, wait_for_block, skip_if_testrp
 
     wait_for_block(web3, current_block + 3)
 
-    with gevent.Timeout(5):
-        while len(seen_blocks) < 2:
-            gevent.sleep(random.random())
+    while len(seen_blocks) < 2:
+        pass
```

Comment:
I don't like that this will hang indefinitely if there is an underlying failure.  Can you re-add the timeout and just increase the wait time?